### ISSUE TITLE
Not highlighting typescript issue in vue files

### ISSUE THREAD

**Issue by rchl** [atom-ide-community/atom-ide-vue#4]
I can provide small repro if needed but maybe it will be clear from my description without that...

Seems like I can see types of objects/properties on hovering them but if there is some type error or object with implicit 'any' type (I have strict: true enabled in my project), then there are no highlights in the editor telling me about the issue. I can see those in SublimeText+LSP or VSCode+vetur.

**Comment by rchl** [comment]
See Atom vs. Vscode:
<img width="784" alt="atom" src="https://user-images.githubusercontent.com/153197/65140927-e6059180-da0f-11e9-8d4a-8fb4b21976ee.png">
<img width="692" alt="vscode" src="https://user-images.githubusercontent.com/153197/65140928-e6059180-da0f-11e9-8ae2-e90c6c8b3d6c.png">

**Comment by Belar** [comment]
Hi, @rchl 
Thanks for the report, but the bad news is that currently `atom-ide-vue` doesn't provide TypeScript support. It has been touched upon in the issue https://github.com/atom-ide-community/atom-ide-vue/issues/3. 

Unfortunately, from what I know, VLS itself (the LSP from Vetur project) doesn't handle TypeScript. In other editors, Sublime and VS Code, plugins (LSP and Vetur) leverage embedded language functionality.

Closing this one as a duplicate, but I have this issue in mind and will keep https://github.com/atom-ide-community/atom-ide-vue/issues/3 updated (feel free to drop a line there); there is no ETA tho.

**Comment by rchl** [comment]
@Belar this is not typescript code but plain javascript in vue file.

VLS handles typescript, VLS is an interface on top of bundled typescript compiler.
Sublime doesn't have any built-in typescript functionality either. It's all done through LSP package and individual servers like VLS.

I think this is a total misunderstanding of what the issue is. Since this package is using the same VLS package (and even the same version) that, for example, Sublime's `LSP-vue` package uses, it should get the same info from it and the issue seems to be in not showing that information in the UI.

Basically those errors marked with squiggly lines should be coming from `publishDiagnostics` message sent from VLS. And I would assume you are getting those, just not showing them in the UI.

**Comment by Belar** [comment]
> this is not typescript code but plain javascript in vue file.

Yes, but those highlights still come from TypeScript's type checking, aren't they? Compiler errors in case of VLS (since there is no linting support).

> VLS handles typescript.

You are correct. I saw the compiler, but misinterpreted LSP client-plugins in Sublime and VS Code, thought there is more multi-LSP management going. After debugging Vetur in VS Code, can see there are type errors coming from VLS.

> Since this package is using the same VLS package (and even the same version) that, for example, Sublime's LSP-vue package uses, it should get the same info from it and the issue seems to be in not showing that information in the UI.
> Basically those errors marked with squiggly lines should be coming from publishDiagnostics message sent from VLS. And I would assume you are getting those, just not showing them in the UI.

Linter is set up to accept those notifications, but there are no records in `textDocument/publishDiagnostics` coming from VLS. Based on the debugging mentioned earlier, it seems like the same project on Atom (atom-ide-vue) doesn't get some notifications present on VS Code (Vetur). Will investigate, should be setup related.

Thanks for pushing on this one, it shed some light on enabling TypeScript support; which would be great.

**Comment by Belar** [comment]
Update on the issue, with good and bad news.

Based on further investigation and troubleshooting, there are 2 factors in missing TS/JS diagnostics. Validation in general, because hints for template (e.g. about `key` for `v-for`) and style are also not working.

1. `atom-ide-vue` is missing `validation` configuration for `vetur`.
2. [VLS restricts validation of documents](https://github.com/vuejs/vetur/blob/master/server/src/services/vls.ts#L450) based on `languageId`, which is expected to be `vue` (not sure about ST, VS Code has dedicated setting for it in its grammars). In Atom (`atom-languageclient`), id is being sourced from `name` in [grammar's settings](https://github.com/atom/atom-languageclient/blob/485bb9d706b422456640c9070eee456ef2cf09c0/lib/adapters/document-sync-adapter.ts#L224); later transformed to lowercase. Which, correctly, based on official syntax highlight (developed for Sublime Text and VS Code), is "Vue Component". `vue component` doesn't match `vue`, so validations are not being performed.

First problem is clearly in `atom-ide-vue`'s scope. Unfortunately, fixing it is not enough, something has to be done regarding `languageId`. Possible solutions are:

1. `name` update in `language-vue` package.
2. Add more reliable language id source to `atom-languageclient`.
3. Roll out grammar for Vue as part of `atom-ide-vue`, control all settings.

Outside option 3, change is up to "upstream" projects. The best would probably be a mix of 1 and 2, like derive id from `scopeName` or having more explicit setting for language's id.

Will update configuration as the first thing, it will at least enable one-line workaround (name edit in `language-vue`). From there we can see how to handle the id issue.

**Comment by Belar** [comment]
With the fix referenced above (released in version 1.1.2), this issue should be resolved.

Short sum up, fourth option "revealed itself", which is dynamic grammar modifications. `atom-ide-vue` will modify grammar's name to work around problem described in the previous post. It seems like a good comprise, which is also fully within the reach of this package.

Happy coding! :tada:
